test/JDBC/expected/BABEL-5242-union-tsql-basetype.out

```diff
@@ -0,0 +1,458 @@
+    select
+        traamo = a
+    from t1
+union all
+    select
+        traamo = b
+    from t1;
+GO
+
+SELECT c.[name] AS [sys_types_user_type_id],
+       TYPE_NAME(a.user_type_id) AS type_name_function_user_type_id
+FROM sys.columns a
+    INNER JOIN sys.objects b
+        ON a.[object_id] = b.[object_id]
+    LEFT JOIN sys.types c
+        ON a.user_type_id = c.user_type_id
+WHERE b.[name] = 'TestView'
+    AND a.[name] = 'traamo';
+GO
+~~START~~
+varchar#!#nvarchar
+money#!#money
+~~END~~
+
+
+DROP VIEW TestView;
+GO
+
+CREATE view TestView as
+    select
+        traamo = b
+    from t1
+union all
+    select
+        traamo = a
+    from t1;
+GO
+
+SELECT c.[name] AS [sys_types_user_type_id],
+       TYPE_NAME(a.user_type_id) AS type_name_function_user_type_id
+FROM sys.columns a
+    INNER JOIN sys.objects b
+        ON a.[object_id] = b.[object_id]
+    LEFT JOIN sys.types c
+        ON a.user_type_id = c.user_type_id
+WHERE b.[name] = 'TestView'
+    AND a.[name] = 'traamo';
+GO
+~~START~~
+varchar#!#nvarchar
+money#!#money
+~~END~~
+
+
+DROP VIEW TestView;
+GO
+
+DROP TABLE t1;
+GO
+
+CREATE TABLE ti (
+  a tinyint,
+  b smallint
+);
+GO
+
+CREATE VIEW TestView AS
+    select
+        traamo = a
+    from ti
+union all
+    select
+        traamo = null
+    from ti;
+GO
+
+-- Should be tinyint instead of smallint
+SELECT c.[name] AS [sys_types_user_type_id],
+       TYPE_NAME(a.user_type_id) AS type_name_function_user_type_id
+FROM sys.columns a
+    INNER JOIN sys.objects b
+        ON a.[object_id] = b.[object_id]
+    LEFT JOIN sys.types c
+        ON a.user_type_id = c.user_type_id
+WHERE b.[name] = 'TestView'
+    AND a.[name] = 'traamo';
+GO
+~~START~~
+varchar#!#nvarchar
+tinyint#!#tinyint
+~~END~~
+
+
+DROP VIEW TestView;
+GO
+
+CREATE VIEW TestView AS
+    select
+        traamo = a
+    from ti
+union all
+    select
+        traamo = b
+    from ti;
+GO
+
+-- Should be smallint since UNION between tinyint and smallint
+SELECT c.[name] AS [sys_types_user_type_id],
+       TYPE_NAME(a.user_type_id) AS type_name_function_user_type_id
+FROM sys.columns a
+    INNER JOIN sys.objects b
+        ON a.[object_id] = b.[object_id]
+    LEFT JOIN sys.types c
+        ON a.user_type_id = c.user_type_id
+WHERE b.[name] = 'TestView'
+    AND a.[name] = 'traamo';
+GO
+~~START~~
+varchar#!#nvarchar
+smallint#!#smallint
+~~END~~
+
+
+DROP VIEW TestView;
+GO
+
+CREATE VIEW TestView AS
+    select
+        traamo = b
+    from ti
+union all
+    select
+        traamo = a
+    from ti;
+GO
+
+SELECT c.[name] AS [sys_types_user_type_id],
+       TYPE_NAME(a.user_type_id) AS type_name_function_user_type_id
+FROM sys.columns a
+    INNER JOIN sys.objects b
+        ON a.[object_id] = b.[object_id]
+    LEFT JOIN sys.types c
+        ON a.user_type_id = c.user_type_id
+WHERE b.[name] = 'TestView'
+    AND a.[name] = 'traamo';
+GO
+~~START~~
+varchar#!#nvarchar
+smallint#!#smallint
+~~END~~
+
+
+INSERT INTO ti VALUES (255, 256), (0, -1);
```

Comment:
we should insert value which will breach precision/scale of smallmoney but fits into money and vice versa

Reply:
Good idea! Supplemented.